# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1223037416 -7200
#      Fri Oct 03 14:36:56 2008 +0200
# Branch release-3-0-x
# Node ID ad3b944fde43e8f4b189ca48650ce1cd9c4a277d
# Parent  9026c7cc22effa8c6694bbdf0ca3ebc55c6f08ac
fplot.m: call axis after plot

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2008-09-24  John W. Eaton  <jwe@octave.org>
+
+	* plot/fplot.m: Call axis after calling plot.
+
 2008-09-24  Jaroslav Hajek  <highegg@gmail.com>
 
 	Version 3.0.3 released.
 
 2008-09-24  John W. Eaton  <jwe@octave.org>
 
 	* general/interpft.m: Increase tolerance in tests.
 
diff --git a/scripts/plot/fplot.m b/scripts/plot/fplot.m
--- a/scripts/plot/fplot.m
+++ b/scripts/plot/fplot.m
@@ -93,25 +93,26 @@ function fplot (fn, limits, n, linespec)
       x = linspace (limits(1), limits(2), n)';
       y = feval (fn, x);
     endwhile 
   else
     x = linspace (limits(1), limits(2), n)';
     y = feval (fn, x);
   endif
 
-  if (length (limits) > 2) 
-    axis (limits);
-  endif
-
   if (have_linespec)
     plot (x, y, linespec);
   else
     plot (x, y);
   endif
+
+  if (length (limits) > 2) 
+    axis (limits);
+  endif
+
   if (isvector(y))
     legend (nam);
   else
     for i=1:columns(y)
       nams{i} = sprintf ("%s(:,%i)", nam, i);
     endfor
     legend (nams{:});
   endif
